KAFKA-16958 add STRICT_STUBS to EndToEndLatencyTest, OffsetCommitCallbackInvokerTest, ProducerPerformanceTest, and TopologyTest

dujian0068 opened the pull request:
[KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`](https://issues.apache.org/jira/browse/KAFKA-16958?jql=project%20%3D%20KAFKA%20AND%20assignee%20in%20(currentUser())%20ORDER%20BY%20created%20DESC)



### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

dujian0068 committed:
KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`

chia7712 commented on streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala:
```diff
@@ -54,6 +42,7 @@ import scala.jdk.CollectionConverters._
  */
 //noinspection ScalaDeprecation
 @Timeout(600)
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
```
There are two `TopologyTest` in code base, and the one which needs strict stub is `org.apache.kafka.streams.TopologyTest`

https://github.com/apache/kafka/blob/trunk/streams/src/test/java/org/apache/kafka/streams/TopologyTest.java#L83

dujian0068 replied:
Thank you
I I have already added

chia7712 replied:
please revert those changes from this `TopologyTest` :)

dujian0068 replied:
I'm sorry, I misunderstood before.

chia7712 reviewed (commented):
@dujian0068 thanks for this PR. one comment is left

dujian0068 committed:
KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`

dujian0068 committed:
KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`

chia7712 commented on streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala:
```diff
@@ -21,30 +21,18 @@ import java.util
 import org.apache.kafka.streams.scala.serialization.Serdes._
 import org.apache.kafka.streams.scala.kstream._
-import org.apache.kafka.streams.{KeyValue, StreamsBuilder => StreamsBuilderJ, StreamsConfig, TopologyDescription}
+import org.apache.kafka.streams.{KeyValue, StreamsConfig, TopologyDescription, StreamsBuilder => StreamsBuilderJ}
```
Could you please revert those unrelated changes?

dujian0068 replied:
Sorry for causing you trouble. 
Already revert

dujian0068 committed:
Merge branch 'trunk' into KAFKA-16958

dujian0068 committed:
KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`

chia7712 commented:
@dujian0068 please fix build error

dujian0068 committed:
KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`

dujian0068 committed:
Merge branch 'trunk' into KAFKA-16958

dujian0068 commented:
@chia7712 
already edited
